src/Microsoft.Health.Fhir.Api/Features/ApiNotifications/ApiNotificationMiddleware.cs

```diff
@@ -62,24 +63,32 @@ public async Task Invoke(HttpContext context)
                 {
                     apiNotification.Latency = timer.GetElapsedTime();
 
-                    IFhirRequestContext fhirRequestContext = _fhirRequestContextAccessor.FhirRequestContext;
-
-                    // For now, we will only emit metrics for audited actions (e.g., metadata will not emit metrics).
-                    if (fhirRequestContext.AuditEventType != null)
+                    try
                     {
-                        apiNotification.Authentication = fhirRequestContext.Principal.Identity.AuthenticationType;
-                        apiNotification.FhirOperation = fhirRequestContext.AuditEventType;
-                        apiNotification.Protocol = context.Request.Scheme;
-                        apiNotification.ResourceType = fhirRequestContext.ResourceType;
-                        apiNotification.StatusCode = (HttpStatusCode)context.Response.StatusCode;
-
-                        await _mediator.Publish(apiNotification, CancellationToken.None);
+                        IFhirRequestContext fhirRequestContext = _fhirRequestContextAccessor.FhirRequestContext;
 
-                        if (fhirRequestContext.StorageRequestMetrics != null)
+                        // For now, we will only emit metrics for audited actions (e.g., metadata will not emit metrics).
+                        if (fhirRequestContext?.AuditEventType != null)
                         {
-                            await _mediator.Publish(fhirRequestContext.StorageRequestMetrics, CancellationToken.None);
+                            apiNotification.Authentication = fhirRequestContext.Principal?.Identity.AuthenticationType;
+                            apiNotification.FhirOperation = fhirRequestContext.AuditEventType;
+                            apiNotification.Protocol = context.Request.Scheme;
+                            apiNotification.ResourceType = fhirRequestContext.ResourceType;
+                            apiNotification.StatusCode = (HttpStatusCode)context.Response.StatusCode;
+
+                            await _mediator.Publish(apiNotification, CancellationToken.None);
+
+                            if (fhirRequestContext.StorageRequestMetrics != null)
+                            {
+                                await _mediator.Publish(fhirRequestContext.StorageRequestMetrics, CancellationToken.None);
+                            }
                         }
                     }
+                    catch (Exception e)
```

Comment:
Why not just check for a null FhirRequestContext and return instead of catching all the exceptions? This seems overly broad and may hide other issues in the future.

Reply:
Need to confirm, but if a NotificationHandler throws, it might bubble up here. The this middleware shouldn't fail here because of this

Reply:
Yeah, looks like an exception in a handler will bubble up to this middleware.